exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt

```diff
@@ -285,6 +285,43 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
         }
     }
 
+    @Suppress("MagicNumber")
+    override fun existingSequences(): List<String> {
+        val sequences = mutableListOf<String>()
+
+        when (currentDialect) {
+            is OracleDialect -> {
+                TransactionManager.current().exec("SELECT SEQUENCE_NAME FROM USER_SEQUENCES") { rs ->
+                    while (rs.next()) {
+                        sequences.add(rs.getString("SEQUENCE_NAME"))
+                    }
+                }
+            }
+            is SQLServerDialect -> {
+                TransactionManager.current().exec("SELECT name FROM sys.sequences") { rs ->
+                    while (rs.next()) {
+                        sequences.add(rs.getString("name"))
+                    }
+                }
+            }
+            is H2Dialect -> {
+                TransactionManager.current().exec("SELECT SEQUENCE_NAME FROM INFORMATION_SCHEMA.SEQUENCES ") { rs ->
+                    while (rs.next()) {
+                        sequences.add(rs.getString("SEQUENCE_NAME"))
+                    }
+                }
+            }
+            else -> {
+                val rs = metadata.getTables(null, null, null, arrayOf("SEQUENCE"))
+                while (rs.next()) {
+                    sequences.add(rs.getString(3))
+                }
+            }
```

Comment:
For the dialects that don't rely on jdbc metadata, should maybe `VendorDialect.existingSequences()` be overriden in each dialect-specific file and the sql logic extracted?
That way only the default vendor function invokes this module's function, so it continues to only contain metadata logic?